lib/fs.ts

```diff
@@ -176,9 +184,14 @@ class FSPath {
 			pendingOptions: null,
 			throttleTime,
 			throttleTimer: null,
+			finish: new Promise(async resolve => {
+				await this.safeWrite(dataFetcher(), options);
+				await this.finishUpdate();
+				resolve();
+			}) as Promise<void>,
```

Comment:
Would it be worth while writing this without involving an explicit Promise? I think you could do it with an immediately invoked async arrow function, although something like this might possibly be less confusing:

          finish: this.writeAndFinish(dataFetcher, options),

    private async writeAndFinish(dataFetcher: () => string | Buffer, options: object) {
       await this.safeWrite(dataFetcher(), options);
       await this.finishUpdate();
    }

Reply:
Either is preferable to using an explicit promise, I'd lean towards async IIFE